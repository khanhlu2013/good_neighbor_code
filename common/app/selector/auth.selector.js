const loginUser = state => state.auth.loginUser;
const isCheckedAuth = state => state.auth.isCheckedAuth;
const isCheckingAuth = state => state.auth.isCheckingAuth;
const isLoggingOut = state => state.auth.isLoggingOut;

const AuthSelector = {
  loginUser,
  isCheckedAuth,
  isCheckingAuth,
  isLoggingOut
};
export default AuthSelector;
