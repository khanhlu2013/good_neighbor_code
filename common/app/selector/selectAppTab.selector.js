export default function selectAppTab(state) {
  return state.selectAppTab;
}
