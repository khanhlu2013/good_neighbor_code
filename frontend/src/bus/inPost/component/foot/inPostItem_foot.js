import React from "react";
import PropTypes from "prop-types";
import User from "../../../../model/user";
import Share from "../../../../model/share";
import PostItemFootStyle from "../../../postComponent/style/postItem_foot_style";
import InPostItemFootApprove from "./inPostItem_foot_approve";
import InPostItemFootShop from "./inPostItem_foot_shop";
import InPostItemFootRequest from "./inPostItem_foot_request";
import { nullOrRequiredValidator } from "../../../../share/util";

function InPostItemFoot(props) {
  const {
    postId,
    loginUser,
    isActive,
    curBorrowShare,
    myRequestShare,
    isRequestingPost,
    isDeleteingShare,
    isAwaringShare,
    isReturningShare,
    onRequestPost,
    onUnRequestPost,
    onAwareApprovePost,
    onReturnPost
  } = props;

  let content;

  if (myRequestShare) {
    content = (
      <InPostItemFootRequest
        isDeleteingShare={isDeleteingShare}
        onUnRequestPost={onUnRequestPost}
        myRequestShareId={myRequestShare.id}
      />
    );
  } else if (
    curBorrowShare !== null &&
    curBorrowShare.borrower.id === loginUser.id
  ) {
    content = (
      <InPostItemFootApprove
        curBorrowShare={curBorrowShare}
        isAwaringShare={isAwaringShare}
        isReturningShare={isReturningShare}
        onAwareApprovePost={onAwareApprovePost}
        onReturnPost={onReturnPost}
      />
    );
  } else if (isActive) {
    content = (
      <InPostItemFootShop
        postId={postId}
        isRequestingPost={isRequestingPost}
        onRequestPost={onRequestPost}
      />
    );
  } else {
    content = <div>Post is no longer active</div>;
  }

  return <PostItemFootStyle>{content}</PostItemFootStyle>;
}
InPostItemFoot.propTypes = {
  postId: PropTypes.string.isRequired,
  loginUser: PropTypes.instanceOf(User).isRequired,
  isActive: PropTypes.bool.isRequired,
  curBorrowShare: nullOrRequiredValidator("object", Share),
  myRequestShare: nullOrRequiredValidator("object", Share),
  isRequestingPost: PropTypes.bool.isRequired,
  isDeleteingShare: PropTypes.bool.isRequired,
  isAwaringShare: PropTypes.bool.isRequired,
  isReturningShare: PropTypes.bool.isRequired,
  onRequestPost: PropTypes.func.isRequired,
  onUnRequestPost: PropTypes.func.isRequired,
  onAwareApprovePost: PropTypes.func.isRequired,
  onReturnPost: PropTypes.func.isRequired
};

export default InPostItemFoot;
