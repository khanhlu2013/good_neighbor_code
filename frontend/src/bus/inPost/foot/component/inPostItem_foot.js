import React from "react";
import PropTypes from "prop-types";
import PostItemFootStyle from "../../../post/component/style/postItem_foot_style";
import InPostItemFootBorrowContainer from "../container/inPostItem_foot_borrow.con";
import InPostItemFootRequestContainer from "../container/inPostItem_foot_request.con";
import InPostItemFootShopContainer from "../container/inPostItem_foot_shop.con";
import { nullOrRequiredValidator } from "../../../../share/util";

function InPostItemFoot(props) {
  const { postId, myRequestShareId, myBorrowShareId, isActivePost } = props;

  let content;

  if (myRequestShareId) {
    content = (
      <InPostItemFootRequestContainer myRequestShareId={myRequestShareId} />
    );
  } else if (myBorrowShareId) {
    content = (
      <InPostItemFootBorrowContainer myBorrowShareId={myBorrowShareId} />
    );
  } else if (isActivePost) {
    content = <InPostItemFootShopContainer postId={postId} />;
  } else {
    content = <div>Post is no longer active</div>;
  }

  return <PostItemFootStyle>{content}</PostItemFootStyle>;
}
InPostItemFoot.propTypes = {
  postId: PropTypes.string.isRequired,
  myRequestShareId: nullOrRequiredValidator("string"),
  myBorrowShareId: nullOrRequiredValidator("string"),
  isActivePost: PropTypes.bool.isRequired
};

export default InPostItemFoot;
