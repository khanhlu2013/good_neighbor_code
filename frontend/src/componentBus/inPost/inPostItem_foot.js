import React from "react";
import PropTypes from "prop-types";
import { User } from "../../model/user";
import { nullOrRequiredValidator } from "../../util";
import { Share } from "../../model/share";
import { InPostItemFootApprove } from "./inPostItem_foot_approve";
import { InPostItemFootRequest } from "./inPostItem_foot_request";
import { InPostItemFootShop } from "./inPostItem_foot_shop";

function InPostItemFoot(props) {
  const {
    postId,
    loginUser,
    isActive,
    curBorrowShare,
    myRequestShare,
    isRequestingPost,
    isDeleteingShare,
    isAwaringShare,
    isReturningShare,
    onCreateShare,
    onDeleteShare,
    onAwareShare,
    onReturnShare
  } = props;

  let content;

  if (myRequestShare) {
    content = (
      <InPostItemFootRequest
        isDeleteingShare={isDeleteingShare}
        onDeleteShare={onDeleteShare}
        myRequestShareId={myRequestShare.id}
      />
    );
  } else if (
    curBorrowShare !== null &&
    curBorrowShare.borrower.id === loginUser.id
  ) {
    content = (
      <InPostItemFootApprove
        curBorrowShare={curBorrowShare}
        isAwaringShare={isAwaringShare}
        isReturningShare={isReturningShare}
        onAwareShare={onAwareShare}
        onReturnShare={onReturnShare}
      />
    );
  } else if (isActive) {
    content = (
      <InPostItemFootShop
        postId={postId}
        isRequestingPost={isRequestingPost}
        onCreateShare={onCreateShare}
      />
    );
  } else {
    content = <div>Post is no longer active</div>;
  }

  return <div className="post-item-foot">{content}</div>;
}
InPostItemFoot.propTypes = {
  postId: PropTypes.string.isRequired,
  loginUser: PropTypes.instanceOf(User).isRequired,
  isActive: PropTypes.bool.isRequired,
  curBorrowShare: nullOrRequiredValidator("object", Share),
  myRequestShare: nullOrRequiredValidator("object", Share),
  isRequestingPost: PropTypes.bool.isRequired,
  isDeleteingShare: PropTypes.bool.isRequired,
  isAwaringShare: PropTypes.bool.isRequired,
  isReturningShare: PropTypes.bool.isRequired,
  onCreateShare: PropTypes.func.isRequired,
  onDeleteShare: PropTypes.func.isRequired,
  onAwareShare: PropTypes.func.isRequired,
  onReturnShare: PropTypes.func.isRequired
};

export { InPostItemFoot };
