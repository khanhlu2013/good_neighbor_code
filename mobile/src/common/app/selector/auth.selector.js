const loginUser = state => state.auth.loginUser;
const isCheckedAuth = state => state.auth.isCheckedAuth;
const isCheckingAuth = state => state.auth.isCheckingAuth;

const AuthSelector = {
  loginUser,
  isCheckedAuth,
  isCheckingAuth
};
export default AuthSelector;
