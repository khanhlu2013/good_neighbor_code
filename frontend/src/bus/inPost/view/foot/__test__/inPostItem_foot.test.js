import React from "react";
import { shallow } from "enzyme";
import InPostItemFootRequestContainer from "../../../container/foot/inPostItem_foot_request.con";
import InPostItemFootBorrowContainer from "../../../container/foot/inPostItem_foot_borrow.con";
import * as helper from "../../../container/foot/inPostItem_foot.selector";
import InPostItemFootWebView from "../inPostItem_foot.webView";

jest.spyOn(helper, "__getRequestOrBorrowShare");

describe("inPostItem_foot", () => {
  it("render non-request and non-approve and non-active post foot correctly", () => {
    const wrap = shallow(
      <InPostItemFootWebView
        postId={"postIdStub"}
        myRequestShareId={null}
        myBorrowShareId={null}
        isActivePost={false}
      />
    );
    expect(wrap).toMatchSnapshot();
  });

  it("render non-request and non-approve and active post foot correctly", () => {
    const wrap = shallow(
      <InPostItemFootWebView
        postId={"postIdStub"}
        myRequestShareId={null}
        myBorrowShareId={null}
        isActivePost={true}
      />
    );
    expect(wrap).toMatchSnapshot();
  });

  it("render request post foot correctly", () => {
    const myRequestShareId = "5bed125a8c57346a4063160d";

    const wrap = shallow(
      <InPostItemFootWebView
        postId={"postIdStub"}
        myRequestShareId={myRequestShareId}
        myBorrowShareId={null}
        isActivePost={true}
      />
    );
    expect(wrap).toMatchSnapshot();
    expect(
      wrap.find(InPostItemFootRequestContainer).props().myRequestShareId
    ).toBe(myRequestShareId);
  });

  it("render approve post foot correctly", () => {
    const myBorrowShareId = "5bed125a8c57346a4063160d";

    const wrap = shallow(
      <InPostItemFootWebView
        postId={"postIdStub"}
        myRequestShareId={null}
        myBorrowShareId={myBorrowShareId}
        isActivePost={true}
      />
    );
    expect(wrap).toMatchSnapshot();
    expect(
      wrap.find(InPostItemFootBorrowContainer).props().myBorrowShareId
    ).toBe(myBorrowShareId);
  });
});
