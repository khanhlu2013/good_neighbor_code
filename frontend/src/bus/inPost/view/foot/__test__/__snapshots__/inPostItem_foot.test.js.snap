// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inPostItem_foot render approve post foot correctly 1`] = `
<styled.div>
  <Connect(InPostItemFootBorrow)
    myBorrowShareId="5bed125a8c57346a4063160d"
  />
</styled.div>
`;

exports[`inPostItem_foot render non-request and non-approve and active post foot correctly 1`] = `
<styled.div>
  <Connect(InPostItemFootShop)
    postId="postIdStub"
  />
</styled.div>
`;

exports[`inPostItem_foot render non-request and non-approve and non-active post foot correctly 1`] = `
<styled.div>
  <div>
    Post is no longer active
  </div>
</styled.div>
`;

exports[`inPostItem_foot render request post foot correctly 1`] = `
<styled.div>
  <Connect(InPostItemFootRequest)
    myRequestShareId="5bed125a8c57346a4063160d"
  />
</styled.div>
`;
